packages/fxa-payments-server/server/lib/server.js

```diff
@@ -250,9 +266,14 @@ module.exports = () => {
 
     INDEX_ROUTES.forEach((route) => {
       // FIXME: should set ETag, Not-Modified:
-      app.get(route, (req, res) => {
+      app.get(route, noCache(), (req, res) => {
```

Comment:
@chenba We added this `nocache` middleware to the index routes. LMK us you are cool with that or not. Thanks!

Reply:
Is it possible to only nocache if we're on this `SubscriptionCreate` Product route? Do we care?

Reply:
According to the `helmet` docs, we can turn off etags with `app.use(helmet.noCache({ noEtag: true }))`.  (`app.use(helmet.noCache())` is also a thing.)

> 
> 
> Is it possible to only nocache if we're on this `SubscriptionCreate` Product route? Do we care?

While I don't like the idea of no caching where it is helpful, I don't think we should worry too much here.  Payments doesn't get _that_ much traffic in the grand scheme of things.

Reply:
yeah, those are my thoughts too. 1) This is probably the least trafficked public website at Mozilla. 2) It is just rendering a static HTML template with a few string replaces. 3) And, you only go through checkout once in a lifetime and will never need to go back. Hopefully :)

Reply:
> According to the helmet docs, we can turn off etags with app.use(helmet.noCache({ noEtag: true })). (app.use(helmet.noCache()) is also a thing.)

So Helmet moved out their `noCache` method to a separate Node package (we tried just using `helmet.noCache()` and it threw a descriptive error). Based on the [source code](https://github.com/helmetjs/nocache/blob/dba4c205867e77be4b55537b0844c7ea9ebc6c5a/index.ts#L3), there are no parameters for the exported function. Would the headers set by `noCache()` change the behavior substantially versus just removing the ETag header?

Reply:
Express won't generate the `Etag` since we are setting the `CacheControl` header to `no-cache`. So it ends up having the same effect.

Reply:
I think simply removing the etag header is fine.  As far as I can tell, we are not currently sending any other caching headers.

Reply:
Per `noCache`'s [README](https://github.com/helmetjs/nocache/blob/dba4c205867e77be4b55537b0844c7ea9ebc6c5a/README.md), they set a bunch of cache-related headers:

> This sets four headers, disabling a lot of browser caching:
> - `Cache-Control: no-store, no-cache, must-revalidate, proxy-revalidate`
> - `Pragma: no-cache`
> - `Expires: 0`
> - `Surrogate-Control: no-store`